pxr/usd/usdUtils/validators.cpp

```diff
@@ -104,11 +104,54 @@ _PackageEncapsulationValidator(const UsdStagePtr& usdStage) {
     return errors;
 }
 
+static
+UsdValidationErrorVector
+_CompressionValidator(const UsdStagePtr& usdStage) {
```

Comment:
@tallytalwar I've come across something in this validator that's not clear to me.

My expectation from reading the description from the existing [CompressionChecker](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdUtils/complianceChecker.py#L149) is that this test should make sure that files within the USDZ file itself were not compressed. So I've created a sample file like this (in this [repo](https://github.com/beersandrew/usd-test-assets/tree/main/Validation/examples/CompressionChecker) as `compressedUsdaInside.usdz`). However this file fails [here](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/sdf/textFileFormat.cpp#L161), before it can get to the `usdchecker` code. 

If I try a different approach of compressing the usdz itself, it also fails early [here](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usd/usdzResolver.cpp#L217).

Do you know how to reproduce the nicer errors from `usdchecker` itself?

Reply:
Also wanted to ping @dgovil to see if maybe he has an example of a failing CompressionChecker file, or [ByteAlignmentChecker](https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/usdUtils/complianceChecker.py#L124) file. As I think I have the code for both of these validators, but am not sure how to get a test file that produces the error in `usdchecker` as of the latest release.